android/src/main/java/com/adyenreactnativesdk/component/dropin/DropInModule.kt

```diff
@@ -195,6 +200,57 @@ class DropInModule(context: ReactApplicationContext?) : BaseModule(context), Rea
         CheckoutProxy.shared.advancedService?.sendRecurringResult(result)
     }
 
+    @ReactMethod
+    fun provideBalance(success: Boolean, balance: ReadableMap?, error: ReadableMap?) {
+        val listener = getService()
+        if (listener == null) {
+            sendErrorEvent(ModuleException.NoModuleListener(integration))
+            return
+        }
+        if (success) {
+            val jsonObject = ReactNativeJson.convertMapToJson(balance)
+            val balanceResult = BalanceResult.SERIALIZER.deserialize(jsonObject)
+            listener.sendBalanceResult(BalanceDropInServiceResult.Balance(balanceResult))
+        } else {
+            val message = error?.getString(AdyenConstants.PARAMETER_MESSAGE)
+            listener.sendBalanceResult(BalanceDropInServiceResult.Error(null, message, true)  )
```

Comment:
Did you leave first parameter (which is the error dialog that will be shown) in `BalanceDropInServiceResult.Error()` as null on purpose? I assume we're not showing any pop ups when this happens in the flow right now. On android we pass an `ErrorDialog` here when something goes wrong to show a pop up. You can take a look at `handleBalanceResponse` function in `ExampleAdvancedDropInService` or `requestBalanceCall` in `SessionDropInService` to see how we do it. The same goes for other kinds of `DropInServiceResult.Error` instances such as order creation etc.

Reply:
Correct. Intent here to dismiss the DropIn and communicate error back to merchant's level and so they can handle UI state (present alert etc)
Another reason - there is no popup on iOS, so dismissing DropIn makes behaviour more align.